src/transport/SecureSessionMgr.cpp

```diff
@@ -266,7 +266,7 @@ void SecureSessionMgrBase::HandleDataReceived(const PacketHeader & packetHeader,
         VerifyOrExit(
             payloadlen <= len,
             (ChipLogError(Inet, "Secure transport can't find MAC Tag; buffer too short"), err = CHIP_ERROR_INVALID_MESSAGE_LENGTH));
-        err = mac.Decode(packetHeader, &data[payloadlen], len - payloadlen, &taglen);
+        err = mac.Decode(packetHeader, &data[payloadlen], static_cast<uint16_t>(len - payloadlen), &taglen);
```

Comment:
this kind of cast makes me wonder if we're going the wrong direction with this.  maybe we should isolate the `uint16_t`-limitations in PacketBuffer and in DM protocol details

Reply:
> this kind of cast makes me wonder if we're going the wrong direction with this

The biggest problem here is that C's integer-promotion rules for arithmetic are kind of dumb.  If those were `unsigned int`, the difference would also be `unsigned int`, but a difference of two `uint16_t` gets promoted to `int`.  Without `-Wconversion` it's not an issue, of course, because the narrowing conversion is silent then...  I wish compilers were smarter here; a sufficiently smart compiler can in fact prove that `len - payloadlen` is a valid `uint16_t` value based on those two lines of code, and could avoid warning about the "narrowing" conversion.  Ah, well.

We _could_ switch a bunch of this stuff to `size_t`, but then we run into the need to do a bunch of casting and/or dynamic checks needed when indexing into `PacketBuffer`s.

I guess I could change the in-progress code I have for https://github.com/project-chip/connectedhomeip/issues/2914 to take `size_t`-sized buffers, not `uint16_t`-sized, so we can keep accepting `size_t`-sized things in the `Decode` methods here.  We get impedance mismatches that way too, though....